include/ifc/abstract-sgraph.hxx

```diff
@@ -3544,7 +3544,18 @@ namespace ifc {
 
     // -- exception type in case of an invalid partition name
     struct InvalidPartitionName {
-        std::string_view name;
+        static constexpr unsigned NAME_BUFFER_SIZE = 64;
+
+        char name[NAME_BUFFER_SIZE]                = "";
+
+        static InvalidPartitionName make(std::string_view name) noexcept
+        {
+            InvalidPartitionName e{};
+            name = name.substr(0, NAME_BUFFER_SIZE - 1);
+            strncpy_s(e.name, name.data(), name.length());
```

Comment:
`strncpy_s` is a C11 function, takes 4 arguments and it's not guaranteed to be available: https://en.cppreference.com/w/c/string/byte/strncpy

> As with all bounds-checked functions, strncpy_s only guaranteed to be available if `__STDC_LIB_EXT1__` is defined by the implementation and if the user defines `__STDC_WANT_LIB_EXT1__` to the integer constant `1` before including [<string.h>](https://en.cppreference.com/w/c/string/byte).

The return value is also not checked.

However, it's very silly to use this function here. This should just be a `std::copy` or `std::memcpy`.

Reply:
ok